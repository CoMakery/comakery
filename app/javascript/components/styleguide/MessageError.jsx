import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Message from './Message'

class MessageError extends React.Component {
  render() {
    const {className, ...other} = this.props

    const classnames = classNames(
      'message__error',
      className
    )

    return (
      <React.Fragment>
        <Message
          className={classnames} {...other}
          severity="error"
        />
      </React.Fragment>
    )
  }
}

MessageError.propTypes = {
  className: PropTypes.string,
  text     : PropTypes.string
}
MessageError.defaultProps = {
  className: '',
  text     : 'error text'
}
export default MessageError
