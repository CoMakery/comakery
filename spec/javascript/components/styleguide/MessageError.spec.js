import React from 'react'
import { shallow, mount, render } from 'enzyme'
import MessageError from 'components/styleguide/MessageError'

describe('MessageError', () => {
    it('renders correctly without props', () => {
      const wrapper = shallow(<MessageError/>)

      expect(wrapper).toMatchSnapshot()
      expect(wrapper.exists('.message__error')).toBe(true)
    })

    it('renders correctly with custom className', () => {
      const wrapper = shallow(<MessageError className='__test' />)

      expect(wrapper.exists('.message__error.__test')).toBe(true)
    })
})
