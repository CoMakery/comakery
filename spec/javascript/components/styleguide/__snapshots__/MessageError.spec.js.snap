// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageError renders correctly without props 1`] = `
<Fragment>
  <Message
    className="message__error"
    severity="error"
    text="error text"
  />
</Fragment>
`;
